Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -8186,10 +8195,44 @@ - (void)_elementDidFocus:(const WebKit::FocusedElementInformation&)information u
 
     [_webView _resetFocusPreservationCount];
 
-    _focusRequiresStrongPasswordAssistance = NO;
     _additionalContextForStrongPasswordAssistance = nil;
+
+    BOOL requiresKeyboard = mayContainSelectableText(information.elementType);
+    _focusedElementInformation = information;
+
     if ([inputDelegate respondsToSelector:@selector(_webView:focusRequiresStrongPasswordAssistance:)])
-        _focusRequiresStrongPasswordAssistance = [inputDelegate _webView:self.webView focusRequiresStrongPasswordAssistance:focusedElementInfo.get()];
+        [self _continueElementDidFocus:[inputDelegate _webView:self.webView focusRequiresStrongPasswordAssistance:focusedElementInfo.get()] focusedElementInfo:focusedElementInfo requiresKeyboard:requiresKeyboard activityStateChanges:activityStateChanges isChangingFocusForScope:WTFMove(isChangingFocusForScope) isFocusingElementWithKeyboardForScope:WTFMove(isFocusingElementWithKeyboardForScope)];
+    else if ([inputDelegate respondsToSelector:@selector(_webView:focusRequiresStrongPasswordAssistance:completionHandler:)]) {
+        auto checker = WebKit::CompletionHandlerCallChecker::create(inputDelegate, @selector(_webView:focusRequiresStrongPasswordAssistance:completionHandler:));
+        [inputDelegate _webView:self.webView focusRequiresStrongPasswordAssistance:focusedElementInfo.get() completionHandler:makeBlockPtr([
+            weakSelf = RetainPtr { self },
+            checker = WTFMove(checker),
+            focusedElementInfo,
+            requiresKeyboard,
+            activityStateChanges,
+            isChangingFocusForScope = WTFMove(isChangingFocusForScope),
+            isFocusingElementWithKeyboardForScope = WTFMove(isFocusingElementWithKeyboardForScope),
+            focusedElementInformationIdentifier = _focusedElementInformation.identifier
+        ] (BOOL result) mutable {
+            if (checker->completionHandlerHasBeenCalled())
+                return;
+            checker->didCallCompletionHandler();
+            RetainPtr strongSelf = weakSelf.get();
+            if (!strongSelf)
+                return;
+            if (focusedElementInformationIdentifier != strongSelf->_focusedElementInformation.identifier)
+                return;
+            [strongSelf _continueElementDidFocus:result focusedElementInfo:focusedElementInfo requiresKeyboard:requiresKeyboard activityStateChanges:activityStateChanges isChangingFocusForScope:WTFMove(isChangingFocusForScope) isFocusingElementWithKeyboardForScope:WTFMove(isFocusingElementWithKeyboardForScope)];
```

Comment:
This would be more readable with some newlines.

Reply:
It'll be more readable with one CompletionHandlerCallingScope